packages/devtools_app/lib/src/framework/theme_manager.dart

```diff
@@ -0,0 +1,94 @@
+import '../shared/globals.dart';
+
+final _log = Logger('theme_manager');
+
+/// Manages changes in theme settings from an editor/IDE.
+class EditorThemeManager extends DisposableController
+    with AutoDisposeControllerMixin {
+  EditorThemeManager(DartToolingDaemon dtd)
+      : editorClient = DtdEditorClient(dtd);
+
+  final DtdEditorClient editorClient;
+
+  void listenForThemeChanges() {
+    autoDisposeStreamSubscription(
+      editorClient.event.listen((event) {
+        if (event is ThemeChangedEvent) {
+          final currentTheme = getIdeTheme();
+          final newTheme = event.theme;
+
+          if (currentTheme.isDarkMode != newTheme.isDarkMode) {
+            updateQueryParameter(
+              IdeThemeQueryParams.devToolsThemeKey,
+              newTheme.isDarkMode
+                  ? IdeThemeQueryParams.darkThemeValue
+                  : IdeThemeQueryParams.lightThemeValue,
+            );
+          }
+
+          if (newTheme.backgroundColor != null) {
+            final newBackgroundColor =
+                tryParseColor(newTheme.backgroundColor!, logger: _log);
+            if (newBackgroundColor != null &&
+                newBackgroundColor != currentTheme.backgroundColor) {
+              updateQueryParameter(
+                IdeThemeQueryParams.backgroundColorKey,
+                _colorAsHex(newBackgroundColor),
+              );
+            }
+          }
+
+          if (newTheme.foregroundColor != null) {
+            final newForegroundColor =
+                tryParseColor(newTheme.foregroundColor!, logger: _log);
+            if (newForegroundColor != null &&
+                newForegroundColor != currentTheme.foregroundColor) {
+              updateQueryParameter(
+                IdeThemeQueryParams.foregroundColorKey,
+                _colorAsHex(newForegroundColor),
+              );
+            }
+          }
+
+          if (newTheme.fontSize != null &&
+              newTheme.fontSize!.toDouble() != currentTheme.fontSize) {
+            updateQueryParameter(
+              IdeThemeQueryParams.fontSizeKey,
+              newTheme.fontSize!.toDouble().toString(),
+            );
+          }
+
+          setGlobal(IdeTheme, getIdeTheme());
+
+          // We are toggling to the opposite theme and then back to force the IDE
+          // to update all theme features.
+          // TODO(https://github.com/flutter/devtools/issues/8366): Clean up so
+          // that preferences controller listens for changes in all theme
+          // features.
+          preferences.toggleDarkModeTheme(!newTheme.isDarkMode);
+          preferences.toggleDarkModeTheme(newTheme.isDarkMode);
+        }
+      }),
+    );
+  }
+
+  String _colorAsHex(Color color) {
+    return (color.r.toInt().toRadixString(16).padLeft(2, '0') +
+            color.g.toInt().toRadixString(16).padLeft(2, '0') +
+            color.b.toInt().toRadixString(16).padLeft(2, '0'))
+        .toUpperCase();
```

Comment:
While tseting this for VS Code, I noticed this isn't working correctly.. The doubles stored in color are from 0-1 but this code assumes 0-255. This currently always computes `000000`. It should multiple by 255, something like this:

https://github.com/flutter/engine/blob/61ad9f199f6b97b47c7388591e517cb415935ec0/lib/ui/painting.dart#L188-L190